apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/McsClaimTransformerV2.java

```diff
@@ -318,19 +154,31 @@ private static List<Claim.DiagnosisComponent> getDiagnosis(RdaMcsClaim claimGrou
         .stream()
         .map(
             diagCode -> {
-              String icdVersion = diagCode.getIdrDiagIcdType().equals("0") ? "10" : "9-cm";
-
-              return new Claim.DiagnosisComponent()
-                  .setSequence(diagCode.getPriority() + 1)
-                  .setDiagnosis(
-                      new CodeableConcept()
-                          .setCoding(
-                              List.of(
-                                  new Coding(
-                                      "http://hl7.org/fhir/sid/icd-" + icdVersion,
-                                      diagCode.getIdrDiagCode(),
-                                      null))));
+              Claim.DiagnosisComponent component;
+
+              if (Strings.isNotBlank(diagCode.getIdrDiagCode())) {
+                String system;
+
+                if (VALID_ICD_TYPES.contains(diagCode.getIdrDiagIcdType())) {
+                  system =
+                      diagCode.getIdrDiagIcdType().equals("0")
+                          ? IcdCode.CODING_SYSTEM_ICD_10_CM
+                          : IcdCode.CODING_SYSTEM_ICD_9;
+                } else {
+                  system = null;
```

Comment:
setting this to null; doesn't that risk an NPE in createCodeableConcept? in checking that method, it attempts to use that codingSystem.

Reply:
The variable never has any invocations on it, it's just passed through to the CodeableConcept object where system is rightfully set to null, thus it will not display a system when it is printed.